packages/eslint-plugin/src/rules/prefer-find.ts

```diff
@@ -0,0 +1,283 @@
+        );
+        return parseIfArrayFilterExpression(lastExpression);
+      }
+
+      if (expression.type === AST_NODE_TYPES.ChainExpression) {
+        return parseIfArrayFilterExpression(expression.expression);
+      }
+
+      // Check if it looks like <<stuff>>(...), but not <<stuff>>?.(...)
+      if (
+        expression.type === AST_NODE_TYPES.CallExpression &&
+        !expression.optional
+      ) {
+        const callee = expression.callee;
+        // Check if it looks like <<stuff>>.filter(...) or <<stuff>>['filter'](...),
+        // or the optional chaining variants.
+        if (callee.type === AST_NODE_TYPES.MemberExpression) {
+          const isBracketSyntaxForFilter = callee.computed;
+          if (
+            isBracketSyntaxForFilter
+              ? getStaticValue(callee.property, globalScope)?.value === 'filter'
+              : callee.property.name === 'filter'
+          ) {
+            const filterNode = callee.property;
+
+            const filteredObjectType = getConstrainedTypeAtLocation(
+              services,
+              callee.object,
+            );
+
+            // As long as the object is an array or an optional chain on a
+            // nullable array, this is an Array.prototype.filter expression.
+            if (isArrayish(filteredObjectType)) {
+              return {
+                isBracketSyntaxForFilter,
+                filterNode,
+              };
+            }
+          }
+        }
+      }
+
+      return undefined;
+    }
+
+    /**
+     * Tells whether the type is a possibly nullable array/tuple or union thereof.
+     */
+    function isArrayish(type: Type): boolean {
+      let isAtLeastOneArrayishComponent = false;
+      for (const unionPart of tsutils.unionTypeParts(type)) {
+        if (
+          tsutils.isIntrinsicNullType(unionPart) ||
+          tsutils.isIntrinsicUndefinedType(unionPart)
+        ) {
+          // ignore
+        } else if (
+          checker.isArrayType(unionPart) ||
+          checker.isTupleType(unionPart)
+        ) {
+          isAtLeastOneArrayishComponent = true;
+        } else {
+          return false;
+        }
+      }
+
+      return isAtLeastOneArrayishComponent;
+    }
+
+    function getObjectIfArrayAtExpression(
+      node: TSESTree.CallExpression,
+    ): TSESTree.Expression | undefined {
+      if (node.arguments.length === 1) {
+        const atArgument = getStaticValue(node.arguments[0], globalScope);
+        if (atArgument != null && isTreatedAsZeroByArrayAt(atArgument.value)) {
+          const callee = node.callee;
+          if (
+            callee.type === AST_NODE_TYPES.MemberExpression &&
+            !callee.optional &&
+            (callee.computed
+              ? getStaticValue(callee.property, globalScope)?.value === 'at'
+              : callee.property.name === 'at')
+          ) {
+            return callee.object;
+          }
+        }
+      }
+
+      return undefined;
```

Comment:
[Style] What do you think about returning early to avoid more indentation?

```ts
if (node.arguments.length !== 1) {
  return undefined;
}

// ...
```

(here and elsewhere)

Tangent: I used to use the phrase _"invert if, reduce nesting"_ more in PR reviews. Now I'd rather make a lint rule for it: https://github.com/JoshuaKGoldberg/create-typescript-app/issues/1267

Reply:
Changed this specific instance.... I think it would be better if you'd be willing to chang the other instances that you're concerned with while merging, though. 

I actually don't subscribe to unqualified use of the guard-clause-if pattern (since I think nested ifs without elses make the happy path clearer), so I don't feel qualified to carry out the changes in a way that you find appealing 😅

Reply:
Haha makes sense, I appreciate the honesty 😄. I'll probably lazy-out and just leave as-is.

Most of the time when I've put this feedback the author either didn't care one way or the other or didn't think of the reduced-nesting approach. Since you actually have an opinion I'm much happier just rolling with it.

Reply:
Sounds good. Sorry for being difficult!

Reply:
I wish _more_ people were _more_ "difficult" in this way. 😄